core/store/src/trie/state_parts.rs

```diff
@@ -361,7 +364,7 @@ impl Trie {
                 let mut iter = children.iter();
                 while let Some((index, child)) = iter.next() {
                     let NodeHandle::Hash(h) = child;
-                    let child = self.retrieve_node(h)?.1;
+                    let child = self.retrieve_node(h)?.unwrap().1;
```

Comment:
I'd add some helper to resolve `None` case and return `StorageError` instead of `unwrap`.

Reply:
I'm getting rid of this in the next part